### DISCUSSION TITLE
Dimensionality Reduction, Animated

### DISCUSSION THREAD

**Discussion by jeroenjanssens** [has2k1/plotnine#857] [2024 Plotnine Contest]
### Authors

Jeroen Janssens

### Links

 [blog](https://jeroenjanssens.com), [github](https://github.com/jeroenjanssens), [linkedin](https://www.linkedin.com/in/jeroenjanssens), [twitter](https://twitter.com/jeroenhjanssens)

### Full description

Understanding your data is key in any data science project. Visualization is useful for this, but can be challenging when the data has a high dimensionality. This includes complex data such as text, images, and sensor measurements. You could create a scatter plot matrix, but this can only show how any two features interact and fails to capture structure across many dimensions.

There’s an entire subfield within machine learning concerned with exactly this challenge: dimensionality reduction. Dimensionality reduction algorithms can help you gain insight into your high-dimensional data and reveal whether there’s any structure. The current state-of-the-art dimensionality reduction algorithm is UMAP, which stands for Uniform Manifold Approximation and Projection for Dimension Reduction.

In this post we're going to apply UMAP to the MNIST dataset, a collection of 70,000 handwritten digits. Each digit is an image of 28 by 28 grayscale pixels, making the dimensionality of the dataset 784. UMAP will embed this dataset into two dimensions, allowing us to understand its structure.

The result of is an animation where each frame is an epoch. On the right are random samples of each digit, to give us an idea of how all the variations look like and to serve as a legend.

https://github.com/jeroenjanssens/umap-animated/assets/1368256/cc7aa083-20db-4fe7-9647-98992676ab3c

[Read more...](
https://github.com/jeroenjanssens/dimensionality-reduction-animated)

### Code repository

https://github.com/jeroenjanssens/dimensionality-reduction-animated

**Comment by has2k1** [comment]
https://github.com/jeroenjanssens/umap-animated/assets/1368256/cc7aa083-20db-4fe7-9647-98992676ab3c

**Comment by EconomiCurtis** [comment]
Ahoy @jeroenjanssens, 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Comment by has2k1** [comment]
@jeroenjanssens, , The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you again for taking part in the contest.
